recipes/gromacs/2020/build.sh

```diff
@@ -115,11 +115,11 @@ function _gromacs_bin_dir() {
         "${PREFIX}/bin.AVX_512/identifyavx512fmaunits" | grep -q '2' && \
         arch='AVX_512'
     ;;
-    *\ avx2\ * | *avx_2*)
+    *\ avx2\ * | *avx2_0*)
```

Comment:
See lines 69 and 74 for the things these changes now match

Reply:
Makes sense.  Do we need to gather a bit more info on the output of this command on OSX to check we are not missing anything between versions or types?  @stain and @douglowe, could you look at the output of the `sysctl -a | grep '^hw.optional\..*` and see how it displays these 4 SIMD types?

Reply:
```
hw.optional.avx1_0: 1
hw.optional.avx2_0: 1
hw.optional.avx512f: 0
hw.optional.avx512cd: 0
hw.optional.avx512dq: 0
hw.optional.avx512bw: 0
hw.optional.avx512vl: 0
hw.optional.avx512ifma: 0
hw.optional.avx512vbmi: 0
```
This is what I've got for these SIMD types. For now I think this should work okay. Would be interesting to see what someone with the new ARM-based mac's get for their setup though.

Reply:
Yes, there's no attempt to support the ARM Macs here yet. I have access to such a Mac mini, though, so could try it some day

Reply:
My colleague got output like `hw.optional.avx1_0: 1` on his older mac

Reply:
that would be me i think, but it's actually a new mac (2020). 

I furthermore get:
```
hw.optional.avx1_0: 1
hw.optional.avx2_0: 1
hw.optional.avx512f: 1
```
but on the other hand, when I do `sysctl -a machdep.cpu.features` I get
```
machdep.cpu.features: FPU VME DE PSE TSC MSR PAE MCE CX8 APIC SEP MTRR PGE MCA CMOV PAT PSE36 CLFSH DS ACPI MMX FXSR SSE SSE2 SS HTT TM PBE SSE3 PCLMULQDQ DTES64 MON DSCPL VMX EST TM2 SSSE3 FMA CX16 TPR PDCM SSE4.1 SSE4.2 x2APIC MOVBE POPCNT AES PCID XSAVE OSXSAVE SEGLIM64 TSCTMR AVX1.0 RDRAND F16C
```
which doesn't include AVX2 or AVX512. I don't understand this discrepancy but maybe has to do with what the hardware supports vs what the OS supports?

Reply:
I think you need to check `leaf7_features` and `extfeatures` too, this is what I get:
```
sysctl -a | grep features
machdep.cpu.features: FPU VME DE PSE TSC MSR PAE MCE CX8 APIC SEP MTRR PGE MCA CMOV PAT PSE36 CLFSH DS ACPI MMX FXSR SSE SSE2 SS HTT TM PBE SSE3 PCLMULQDQ DTES64 MON DSCPL VMX SMX EST TM2 SSSE3 FMA CX16 TPR PDCM SSE4.1 SSE4.2 x2APIC MOVBE POPCNT AES PCID XSAVE OSXSAVE SEGLIM64 TSCTMR AVX1.0 RDRAND F16C
machdep.cpu.leaf7_features: RDWRFSGS TSC_THREAD_OFFSET SGX BMI1 HLE AVX2 SMEP BMI2 ERMS INVPCID RTM FPU_CSDS MPX RDSEED ADX SMAP CLFSOPT IPT MDCLEAR TSXFA IBRS STIBP L1DF SSBD
machdep.cpu.extfeatures: SYSCALL XD 1GBPAGE EM64T LAHF LZCNT PREFETCHW RDTSCP TSCI
```

Reply:
Ah, i have AVX512F in my leaf7_features:
```
machdep.cpu.leaf7_features: RDWRFSGS TSC_THREAD_OFFSET SGX BMI1 AVX2 FDPEO SMEP BMI2 ERMS INVPCID FPU_CSDS AVX512F AVX512DQ RDSEED ADX SMAP AVX512IFMA CLFSOPT IPT AVX512CD SHA AVX512BW AVX512VL AVX512VBMI UMIP PKU GFNI VAES VPCLMULQDQ AVX512VNNI AVX512BITALG AVX512VPOPCNTDQ RDPID SGXLC FSREPMOV MDCLEAR IBRS STIBP L1DF ACAPMSR SSBD
```

but when I try running the `<path-of-conda-env>/bin.AVX_512/gmx` executable it outputs
```
Compiled SIMD: AVX_512, but for this host/run AVX2_256 might be better (see
log).
```

Reply:
AVX512F appeared in leaf7_features at https://github.com/golang/go/issues/42649

I don't think we need any of this though!

Reply:
> but when I try running the `<path-of-conda-env>/bin.AVX_512/gmx` executable it outputs
> 
> ```
> Compiled SIMD: AVX_512, but for this host/run AVX2_256 might be better (see
> log).
> ```

Yes, that's what line 115 of build.sh is also checking. Intel didn't (forgot to?) add CPUID bits for the number of FMA units in the modern AVX512-grade hardware. High-end Xeon server chips have 2 and then it's worthwhile. On other stuff, AVX2_256 is faster when using both halves of the lanes separately because the load on the single FMA unit is lower. GROMACS has some custom code to tell these apart - stand back, here be demons!